My Plan: Add Checklist CTA

sirreal opened the pull request:
Add a checklist CTA to the My Plan header based on https://github.com/Automattic/jetpack/issues/11997#issuecomment-494155147

This introduces _yet another primary button_ to this view. https://github.com/Automattic/jetpack/pull/12430 removes the primary buttons from all of the feature cards in the view.

#### Screens

<details>

<summary>Expand for screens</summary>

![placeholder](https://user-images.githubusercontent.com/841763/58160536-f59de500-7c7e-11e9-9177-45b9dc4c99a3.png)
![free-mob](https://user-images.githubusercontent.com/841763/58160542-f8003f00-7c7e-11e9-94cf-850268c12b79.png)
![free-desk](https://user-images.githubusercontent.com/841763/58160551-fafb2f80-7c7e-11e9-9ac7-031195beec7d.png)
![personal-mob](https://user-images.githubusercontent.com/841763/58160554-fe8eb680-7c7e-11e9-9f9a-c4c24bb47815.png)
![personal-desk](https://user-images.githubusercontent.com/841763/58160557-ffbfe380-7c7e-11e9-886a-904dd1772f3f.png)
![premium-mob](https://user-images.githubusercontent.com/841763/58160564-02223d80-7c7f-11e9-9eb8-e60571a410b2.png)
![premium-desk](https://user-images.githubusercontent.com/841763/58160570-03ec0100-7c7f-11e9-9b45-d82e23d0aaf7.png)
![pro-mob](https://user-images.githubusercontent.com/841763/58160574-064e5b00-7c7f-11e9-92b3-5465f3128475.png)
![pro-desk](https://user-images.githubusercontent.com/841763/58160577-077f8800-7c7f-11e9-9e5a-b807c26a6163.png)


</details>

#### Testing instructions:

* From `/wp-admin/admin.php?page=jetpack#/my-plan` on a Jetpack site
* You should see the checklist CTA button like in the above screens.
* For sites with different plans, does the button appear correctly? Does the link work as expected?

#### Proposed changelog entry for your changes:

none

sirreal committed:
Add static Checklist CTA to my plan

sirreal committed:
Remove instead of placeholder

jeherve commented:
Should we show the button on free plans as well? It seems odd, since there isn't much to set up at that point, you are just invited to purchase a plan?

simison commented:
>Should we show the button on free plans as well? It seems odd, since there isn't much to set up at that point, you are just invited to purchase a plan?

There will be some new stuff coming to the checklist sometime after 7.4 release for free sites as well; lazy loading images & CDN for example.

jeffgolenski commented:
We definitely want to display this for free. As @simison mentioned, more things are coming to the checklist soon!

@sirreal I just testing this - I'm noticing a difference in styles. For paid plans the button is in the correct place, but but free it is not. 

Looks great on mobile. Can you just bump the `p` text size to 14px?

![Screen Shot 2019-05-21 at 10 47 03 AM](https://user-images.githubusercontent.com/214813/58106425-15b5a180-7bb6-11e9-96bb-a4ca3e982846.png)
![Screen Shot 2019-05-21 at 10 44 57 AM](https://user-images.githubusercontent.com/214813/58106426-15b5a180-7bb6-11e9-899e-9405053de2be.png)

simison commented:
Should we address multiple primary color buttons on the same page as a follow up to this?

simison commented on _inc/client/my-plan/my-plan-header/checklist-cta.js:
```diff
@@ -0,0 +1,23 @@
+ */
+import Button from 'components/button';
+
+export default function ChecklistCta( { placeholder, siteSlug } ) {
```
is `placeholder` prop used anywhere?

simison replied:
Looks like usage of it was removed in commit 27af4f6e5. Let's clean up here, too, if it's un-used.

sirreal replied:
👍 cf369ad324dcfdb1a56905e36f1a9a6c5f95bd0e

simison reviewed (dismissed):
Code is looking good :thumbsup:

Just small `placeholder` change I'd like to see before merging.

sirreal committed:
Finish removing placeholder references

sirreal committed:
Fix bad class name

sirreal commented on _inc/client/my-plan/my-plan-header/index.js:
```diff
@@ -86,7 +90,7 @@ class MyPlanHeader extends React.Component {
 							/>
 						</div>
-						<div className="jp-landing__plan-iconcard-current">
+						<div className="jp-landing__plan-card-current">
```
This one class was inconsistent with the rest.

sirreal committed:
Remove unused style

sirreal committed:
Remove unused jp-landing-plans__header-description

sirreal committed:
Remove unused jp-landing-plans__header-btn-container class

sirreal committed:
Remove unused jp-landing-plans__header-title class

sirreal committed:
Remove unused jp-landing-plans__header-subtitle class

sirreal committed:
Remove unused classes

sirreal committed:
Remove unused classes

sirreal committed:
Ensure correct line break and placeholder display

sirreal commented:
Rebased so the primary CTAs are no longer repeated (#12430).

> I just testing this - I'm noticing a difference in styles. For paid plans the button is in the correct place, but but free it is not.

There was a misnamed class in exactly 1 place, which was the class I'd based the changes on. I fixed this class name so the different plans should be consistent.

This also highlighted a lot of what appear to be leftover classes from previous plans pages. I've removed many rules that appear to be unused after searching for references.

> Looks great on mobile. Can you just bump the `p` text size to 14px?

I've taken care of that 👍 

<details>

<summary>Here are updated mobile/desktop screenshots for all plans</summary>

![placeholder](https://user-images.githubusercontent.com/841763/58160536-f59de500-7c7e-11e9-9177-45b9dc4c99a3.png)
![free-mob](https://user-images.githubusercontent.com/841763/58160542-f8003f00-7c7e-11e9-94cf-850268c12b79.png)
![free-desk](https://user-images.githubusercontent.com/841763/58160551-fafb2f80-7c7e-11e9-9ac7-031195beec7d.png)
![personal-mob](https://user-images.githubusercontent.com/841763/58160554-fe8eb680-7c7e-11e9-9f9a-c4c24bb47815.png)
![personal-desk](https://user-images.githubusercontent.com/841763/58160557-ffbfe380-7c7e-11e9-886a-904dd1772f3f.png)
![premium-mob](https://user-images.githubusercontent.com/841763/58160564-02223d80-7c7f-11e9-9eb8-e60571a410b2.png)
![premium-desk](https://user-images.githubusercontent.com/841763/58160570-03ec0100-7c7f-11e9-9b45-d82e23d0aaf7.png)
![pro-mob](https://user-images.githubusercontent.com/841763/58160574-064e5b00-7c7f-11e9-92b3-5465f3128475.png)
![pro-desk](https://user-images.githubusercontent.com/841763/58160577-077f8800-7c7f-11e9-9e5a-b807c26a6163.png)


</details>

ockham commented on _inc/client/my-plan/my-plan-header/checklist-cta.js:
```diff
@@ -0,0 +1,20 @@
+	return (
+		<div className="jp-landing__plan-features-header-checklist-cta-container">
+			<Button primary href={ `https://wordpress.com/plans/my-plan/${ siteSlug }` }>
+				{ __( 'View your setup checklist' ) }
```
Is the `__` alias going to work fine here? :thinking: It will probably be handled by the plugin repo's string extraction handling, which is made for `@wordpress/i18n`, and the string will end up in the translation files generated by the plugin repo. But `i18n-calypso` won't load those, will it?

sirreal replied:
This is nothing new or novel, it's how things are done around here 🙂 

https://github.com/Automattic/jetpack/blob/510a46bda82d61245e10f392933a09d839218871/_inc/client/my-plan/my-plan-body.jsx#L8
https://github.com/Automattic/jetpack/blob/510a46bda82d61245e10f392933a09d839218871/_inc/client/my-plan/my-plan-body.jsx#L129

simison replied:
I think there's a script somewhere which pulls these out to a php file, and we're not relying on wp.org process to extract these. 🤔 @zinigor might know better.

jeherve replied:
The magic happens here:
https://github.com/Automattic/jetpack/blob/7b6c7495020716d3d89fd7b0b8d0f4f3093348bb/gulpfile.babel.js#L204

ockham replied:
TIL

ockham reviewed (commented):
One concern [here](https://github.com/Automattic/jetpack/pull/12429/files#r286427482). Appears to be working well otherwise! I also couldn't find any other uses of those stale class names.

Feel free to ship after addressing the i18n question!

jeherve commented on _inc/client/my-plan/my-plan-header/checklist-cta.js:
```diff
@@ -0,0 +1,20 @@
+export default function ChecklistCta( { siteSlug } ) {
+	return (
+		<div className="jp-landing__plan-features-header-checklist-cta-container">
+			<Button primary href={ `https://wordpress.com/plans/my-plan/${ siteSlug }` }>
```
Do we want to add tracking to clicks on that button?

jeherve reviewed (approved):
This looks good, and tests well for me. I think this should be good to merge after a design review.

I mentioned the lack of tracking on that button below, but I don't think that's blocking here, given that we are missing tracking on other buttons in there. It could be added in a future PR I think.

jeffgolenski commented:
>  mentioned the lack of tracking on that button below, but I don't think that's blocking here, given that we are missing tracking on other buttons in there. It could be added in a future PR I think.

Thanks for bringing this up. Yes, we need be adding tracking to all new buttons that are added, every single time. If we don't know if / how our customers are using what we design and build then we're just wasting our time. Can we get a tracks event in there @sirreal? Thank ya! 

cc  @jeherve @simison @ockham

sirreal commented:
> Can we get a tracks event in there @sirreal? 

@jeffgolenski yes, but I'd prefer to add it in a follow up since this has the approvals already. Can we consider this design approved?

jeffgolenski reviewed (approved):
Did another run through. Looks great!

sirreal commented:
Tracks here: https://github.com/Automattic/jetpack/pull/12449
